File: ch19p652SimplestRestorationExample2/ch19p652SimplestRestorationExample2/PresentedViewController.m


#import "PresentedViewController.h"

@interface PresentedViewController ()

@end

@implementation PresentedViewController

-(void)encodeRestorableStateWithCoder:(NSCoder *)coder {
    [super encodeRestorableStateWithCoder:coder];
    NSLog(@"%@ encode %@", self, coder);
}

-(void)decodeRestorableStateWithCoder:(NSCoder *)coder {
    [super decodeRestorableStateWithCoder:coder];
    NSLog(@"%@ decode %@", self, coder);
}

-(void)applicationFinishedRestoringState {
    NSLog(@"finished %@", self);
}

-(void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"view did load %@", self);
    self.view.backgroundColor = [UIColor blueColor];
    UIButton* button = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [button setTitle:@"Pop" forState:UIControlStateNormal];
    [button addTarget:self
               action:@selector(doDismiss:)
     forControlEvents:UIControlEventTouchUpInside];
    [button sizeToFit];
    button.backgroundColor = [UIColor whiteColor];
    button.center = self.view.center;
    [self.view addSubview:button];
}

-(void)doDismiss:(id)sender {
    [self dismissViewControllerAnimated:YES completion:nil];
}



@end

File: iOS8bookExamples/bk2ch06p347SimplestRestorationExample2/ch19p652SimplestRestorationExample2/PresentedViewController.swift

import UIKit

class PresentedViewController : UIViewController {
    
    override func encodeRestorableStateWithCoder(coder: NSCoder) {
        println("\(self) encode \(coder)")
    }
    
    override func decodeRestorableStateWithCoder(coder: NSCoder) {
        println("\(self) decode \(coder)")
    }
    
    override func applicationFinishedRestoringState() {
        println("finished \(self)")
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        println("view did load \(self)")
        self.view.backgroundColor = UIColor.blueColor()
        let button = UIButton.buttonWithType(.System) as! UIButton
        button.setTitle("Dismiss", forState:.Normal)
        button.addTarget(self,
            action:"doDismiss:",
            forControlEvents:.TouchUpInside)
        button.sizeToFit()
        button.center = self.view.center
        button.backgroundColor = UIColor.whiteColor()
        self.view.addSubview(button)
    }
    
    func doDismiss(sender:AnyObject?) {
        self.presentingViewController!.dismissViewControllerAnimated(true, completion: nil)
    }
    
}
